custodia/server/__init__.py

```diff
@@ -81,12 +81,12 @@ def parse_config(cfgfile):
                 config.get('auditlog', 'custodia.audit.log'))
 
             url = config.get('server_url')
-            if url and 'server_socket' in config:
-                raise ValueError(
-                    "'server_url' and ''server_socket'' are mutual exclusive")
-            if url is None:
-                server_socket = os.path.abspath(
-                    config.get('server_socket', 'server_socket'))
+            sock = config.get('server_socket')
+            if not bool(url) ^ bool(sock):
```

Comment:
I get it, the rest was left as an exercise for the reader.
Here is what I want to see in pseudo code:
if not url and not sock:
  raise "A server URL or Server Socket must be specified.
if url and sock:
  raise "Only one of sever_soket or server_url can be specified"

Your code is mathematically correct, but difficult to understand both in code and in the error message.